Entity/Player/Player.cpp

```diff
@@ -0,0 +1,32 @@
+//
+// Created by benek on 10/14/24.
+//
+
+#include "Player.h"
+
+#include "../input.h"
+#include "PlayerStates.h"
+#include <SDL.h>
+
+void Player::Update() {
+
+	if (KeyPressed(SDLK_w)) {
+		m_pStateMachine->ChangeState(PlayerRunning::Instance());
+	}
+
+	if (KeyPressed(SDLK_s)) {
+		m_pStateMachine->ChangeState(PlayerIdle::Instance());
+	}
+
+	if (KeyPressed(SDLK_SPACE)) {
+		m_pStateMachine->ChangeState(PlayerAttacking::Instance());
+	}
+	m_pStateMachine->Update();
+}
```

Comment:
i am curious what you have in mind about input handling? is this okeyish?

Reply:
Yup. For now this is fine!